management/hostedservice/client.go

```diff
@@ -135,3 +136,25 @@ func (self HostedServiceClient) createHostedServiceDeploymentConfig(dnsName, loc
 	}
 	return deployment
 }
+
+func (self HostedServiceClient) AddCertificate(dnsName string, certData []byte, certificateFormat CertificateFormat, password string) (string, error) {
+	if dnsName == "" {
+		return "", fmt.Errorf(errParamNotSpecified, "dnsName")
+	}
+
+	base64CertData := base64.StdEncoding.EncodeToString(certData)
```

Comment:
well normally `data` is used to referred to []bytes (as in above). Once you encode, this is `base64Cert` or better: `certBase64`